core/src/tools/bsmtp.cc

```diff
@@ -546,35 +508,34 @@ int main(int argc, char* argv[])
    *   in them already, we do not enclose the string in < >, otherwise
    *   we do.
    */
-  GetResponse(); /* banner */
-  chat("HELO %s\r\n", my_hostname);
-  chat("MAIL FROM:%s\r\n", cleanup_addr(from_addr, buf, sizeof(buf)));
+  GetResponse(mailhost); /* banner */
+  chat(my_hostname, mailhost, "HELO %s\r\n", my_hostname);
+  chat(my_hostname, mailhost, "MAIL FROM:%s\r\n",
+       cleanup_addr(from_addr, buf, sizeof(buf)));
 
-  for (i = 0; i < argc; i++) {
-    Dmsg1(20, "rcpt to: %s\n", argv[i]);
-    chat("RCPT TO:%s\r\n", cleanup_addr(argv[i], buf, sizeof(buf)));
+  for (const auto& recipient : recipients) {
+    Dmsg1(20, "rcpt to: %s\n", recipient.c_str());
+    chat(my_hostname, mailhost, "RCPT TO:%s\r\n",
+         cleanup_addr(recipient, buf, sizeof(buf)));
   }
 
-  if (cc_addr) {
-    chat("RCPT TO:%s\r\n", cleanup_addr(cc_addr, buf, sizeof(buf)));
+  if (!cc_addr.empty()) {
+    chat(my_hostname, mailhost, "RCPT TO:%s\r\n",
+         cleanup_addr(cc_addr, buf, sizeof(buf)));
   }
   Dmsg0(20, "Data\n");
-  chat("DATA\r\n");
+  chat(my_hostname, mailhost, "DATA\r\n");
 
   //  Send message header
-  fprintf(sfp, "From: %s\r\n", from_addr);
-  Dmsg1(10, "From: %s\r\n", from_addr);
-  if (subject) {
-    fprintf(sfp, "Subject: %s\r\n", subject);
-    Dmsg1(10, "Subject: %s\r\n", subject);
-  }
-  if (reply_addr) {
-    fprintf(sfp, "Reply-To: %s\r\n", reply_addr);
-    Dmsg1(10, "Reply-To: %s\r\n", reply_addr);
+  fprintf(sfp, "From: %s\r\n", from_addr.c_str());
+  Dmsg1(10, "From: %s\r\n", from_addr.c_str());
+  if (!subject.empty()) {
+    fprintf(sfp, "Subject: %s\r\n", subject.c_str());
+    Dmsg1(10, "Subject: %s\r\n", subject.c_str());
   }
-  if (err_addr) {
-    fprintf(sfp, "Errors-To: %s\r\n", err_addr);
-    Dmsg1(10, "Errors-To: %s\r\n", err_addr);
+  if (!reply_addr.empty()) {
+    fprintf(sfp, "Reply-To: %s\r\n", reply_addr.c_str());
+    Dmsg1(10, "Reply-To: %s\r\n", reply_addr.c_str());
   }
 
 #if defined(HAVE_WIN32)
```

Comment:
Note for the future: it is possible that soon or later having a sender not being equal to from will add spam weight to mail and as such make the mail undelivered.

Reply:
@arogge has something to add here or sender

Reply:
we should think about sending the e-mail with the null-sender, as we never want a bounce.

Reply:
@arogge can you have a review if 610150a is ok for that.